firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/CustomModel.java

```diff
@@ -42,7 +44,46 @@
    */
   public CustomModel(
       @NonNull String name, long downloadId, long fileSize, @NonNull String modelHash) {
-    this(name, downloadId, fileSize, modelHash, "");
+    this(name, downloadId, fileSize, modelHash, "", "", 0);
+  }
+
+  /**
+   * Use when creating a custom model from a stored model with a new download is the background.
+   *
+   * @param name - model name
+   * @param downloadId - Android Download Manger - download id
+   * @param fileSize - model file size
+   * @param modelHash - model hash size
+   * @hide
+   */
+  public CustomModel(
+      @NonNull String name,
+      long downloadId,
+      long fileSize,
+      @NonNull String modelHash,
+      String localFilePath) {
+    this(name, downloadId, fileSize, modelHash, localFilePath, "", 0);
+  }
+
+  /**
+   * Use when creating a custom model from a download service response. Download url and download
+   * url expiry should go together. These will not be stored in user preferences as this is a
+   * temporary step towards setting the actual download id.
+   *
+   * @param name - model name
+   * @param modelHash - model hash size
+   * @param fileSize - model file size
+   * @param downloadUrl - download url path
+   * @param downloadUrlExpiry - time download url path expires
+   * @hide
+   */
+  public CustomModel(
```

Comment:
Although probably not a big deal, but the order of common args in this constructor and in the prev one doesn't match.

Reply:
Good catch - made more consistent.